src/grammar.coffee

```diff
@@ -135,36 +135,35 @@ grammar =
 
   # A literal identifier, a variable name or property.
   Identifier: [
-    o 'IDENTIFIER',                             -> new Literal $1
+    o 'IDENTIFIER',                             -> new IdentifierLiteral $1
   ]
 
   # Alphanumerics are separated from the other **Literal** matchers because
   # they can also serve as keys in object literals.
   AlphaNumeric: [
-    o 'NUMBER',                                 -> new Literal $1
+    o 'NUMBER',                                 -> new NumberLiteral $1
```

Comment:
```
Block
  Value IdentifierLiteral: Infinity
```

Hehe.

CoffeeScript actually allows `Infinity = 0`. (That's not valid in strict mode.) Not sure if it's intentional.

Don't really know what to do now.